akka-docs/rst/java/code/docs/http/javadsl/WebSocketClientExampleTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (C) 2016 Typesafe Inc. <http://www.typesafe.com>
+ */
+package docs.http.javadsl;
+
+import akka.Done;
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.model.StatusCodes;
+import akka.http.javadsl.model.headers.Authorization;
+import akka.http.javadsl.model.ws.Message;
+import akka.http.javadsl.model.ws.TextMessage;
+import akka.http.javadsl.model.ws.WebSocketRequest;
+import akka.http.javadsl.model.ws.WebSocketUpgradeResponse;
+import akka.japi.Pair;
+import akka.japi.function.Procedure;
+import akka.stream.ActorMaterializer;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import akka.stream.javadsl.Keep;
+import akka.stream.javadsl.Sink;
+import akka.stream.javadsl.Source;
+
+import java.util.concurrent.CompletionStage;
+
+@SuppressWarnings("unused")
+public class WebSocketClientExampleTest {
+
+  // compile only test
+  public void testSingleWebSocketRequest() {
```

Comment:
Reminds me that it should be possible to define an `@CompileTestOnly` Java annotation that extends also has the `@SuppressWarnings("unused")` annotation to achieve the same what we need here.